core/tee/tee_svc.c

```diff
@@ -110,44 +110,118 @@ uint32_t tee_svc_sys_nocall(void)
+	 * Can be tampered by the REE
+	 * The implementation must still guarantee that the system time
 	 * is monotonous, i.e., successive calls to TEE_GetSystemTime must
 	 * return increasing values of the system time.
+	 * 1000: System time based on a TEE-controlled secure timer.
+	 * The REE cannot interfere with the system time. It may still
+	 * interfere with the scheduling of TEE tasks, but is not able to
+	 * hide delays from a TA calling TEE_GetSystemTime.
+	 */
+	uint32_t sys_time_prot_lvl = 100;
+
+	/*
+	 * TA persistent time protection level
+	 * 100: Persistent time based on an REE-controlled real-time clock
+	 * and on the TEE Trusted Storage for the storage of origins (default).
+	 * 1000: Persistent time based on a TEE-controlled real-time clock
+	 * and the TEE Trusted Storage.
+	 * The real-time clock MUST be out of reach of software attacks
+	 * from the REE.
+	 */
+	uint32_t ta_time_prot_lvl = 100;
+
+	/* Elliptic Curve Cryptographic support (false by default) */
+	bool crypto_ecc_en;
+
+	/*
+	 * Trusted storage anti rollback protection level
+	 * 0 (or missing): No antirollback protection (default)
+	 * 100: Antirollback enforced at REE level
+	 * 1000: Antirollback TEE-controlled hardware
 	 */
-	static const uint32_t sys_time_prot_lvl = 100;
-	static const uint32_t ta_time_prot_lvl = 100;
+	uint32_t ts_antiroll_prot_lvl;
+
+	/* Trusted OS implementation version */
+	char trustedos_impl_version[] = CFG_TEE_IMPL_VERSION;
+
+	/* Trusted OS implementation version (binary value) */
+	uint32_t trustedos_impl_bin_version; /* zero length value by default */
+
+	/* Trusted OS implementation manufacturer name */
+	char trustedos_manufacturer[] = CFG_TEE_MANUFACTURER;
+
+	/* Trusted firmware version */
+	char fw_impl_version[] = CFG_TEE_FW_IMPL_VERSION;
+
+	/* Trusted firmware version (binary value) */
+	uint32_t fw_impl_bin_version; /* zero length value by default */
+
+	/* Trusted firmware manufacturer name */
+	char fw_manufacturer[] = CFG_TEE_FW_MANUFACTURER;
+
+	/* Client identifier */
+	TEE_Identity client_id;
+
+	/* Trusted Application identifier */
+	TEE_UUID ta_app_id;
+
 	struct tee_ta_session *sess;
 	TEE_Result res;
 
+	struct tee_props {
+		void *data;
+		size_t len;
+	};
+
+	/* Consistent with enum utee_property */
+	const struct tee_props tee_props_lut[] = {
+		{api_vers, sizeof(api_vers)},
+		{descr, sizeof(descr)},
+		{&dev_id, sizeof(dev_id)},
+		{&sys_time_prot_lvl, sizeof(sys_time_prot_lvl)},
+		{&ta_time_prot_lvl, sizeof(ta_time_prot_lvl)},
+		{&crypto_ecc_en, sizeof(crypto_ecc_en)},
+		{&ts_antiroll_prot_lvl, sizeof(ts_antiroll_prot_lvl)},
+		{trustedos_impl_version, sizeof(trustedos_impl_version)},
+		{&trustedos_impl_bin_version,
+			sizeof(trustedos_impl_bin_version)},
+		{trustedos_manufacturer, sizeof(trustedos_manufacturer)},
+		{fw_impl_version, sizeof(fw_impl_version)},
+		{&fw_impl_bin_version, sizeof(fw_impl_bin_version)},
+		{fw_manufacturer, sizeof(fw_manufacturer)},
+		{&client_id, sizeof(client_id)},
```

Comment:
I find the current implementation error-prone as, in `case UTEE_PROP_CLIENT_ID:`  we are relying on `sizeof(client_id)` but we are copying somewhere else.

I would suggest to have `{0, 0}` for the client_id, and have the later `case UTEE_PROP_CLIENT_ID:` unchanged (that is uses `sizeof(TEE_Identity)` instead of `tee_props_lut[prop].len`.

The same applies to `UTEE_PROP_TA_APP_ID`

`{0, 0}` can also be used `UTEE_PROP_TEE_DEV_ID`